cardinal/ecs/ecs_test.go

```diff
@@ -469,25 +473,35 @@ func TestCannotSetComponentThatDoesNotBelongToEntity(t *testing.T) {
 	assert.NilError(t, world.RegisterComponents(alpha, beta))
 	assert.NilError(t, world.LoadGameState())
 
-	id, err := world.Create(alpha)
+	id, err := ecs.Create(world, EnergyComponentAlpha{})
 	assert.NilError(t, err)
 
 	err = ecs.SetComponent[EnergyComponentBeta](world, id, &EnergyComponentBeta{100, 200})
 	//err = beta.Set(world, id, EnergyComponentBeta{100, 200})
 	assert.Check(t, err != nil)
 }
 
+type A struct{}
+type B struct{}
+type C struct{}
+type D struct{}
+
+func (A) Name() string { return "a" }
+func (B) Name() string { return "b" }
+func (C) Name() string { return "c" }
+func (D) Name() string { return "d" }
+
 func TestQueriesAndFiltersWorks(t *testing.T) {
 	world := ecs.NewTestWorld(t)
-	a, b, c, d := ecs.NewComponentType[int]("a"), ecs.NewComponentType[int]("b"), ecs.NewComponentType[int]("c"), ecs.NewComponentType[int]("d")
+	a, b, c, d := ecs.NewComponentType[A]("a"), ecs.NewComponentType[B]("b"), ecs.NewComponentType[C]("c"), ecs.NewComponentType[D]("d")
```

Comment:
This is out of scope for this PR, but now that Name() is defined on these structs, is that first parameter to NewComponentType still required?

Reply:
No. This will be removed in a subsequent PR.